analyzers/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -67,6 +68,10 @@ protected enum ErrorLocation
         protected void CheckAndExpression(SyntaxNodeAnalysisContext context)
         {
             var binaryExpression = (TBinaryExpression)context.Node;
+            if (IsInsideTernaryWithThrowExpression(binaryExpression))
```

Comment:
This if can be merged with the one below.

Reply:
Agree. It might break coverage because more conditions will need to be supported. But we don't need to please the tool.